### PR TITLE
Rws Logger

### PR DESCRIPTION
**Pull request by chesuch**
Log rws soap request and response to file

### CODE REVIEW

**Comment by chesuch** [comment]
Logs to output file: var/log/rws.log
RWS credentials are masked in the output.

Example AddOrder output logged to file:

`[2018-04-10 07:29:50] customLogHandler.INFO: <?xml version="1.0" encoding="UTF-8"?>
<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ns1="http://schemas.redarrow.co.uk/rialto" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:ns2="rialto.emerymailorder.com/service.asmx"><SOAP-ENV:Body><ns2:AddRialtoOrders><ns1:rialtoweborders><ns1:orders><ns1:order><ns1:orderheader OrderNo="000000003" OrderDate="2018-03-29T13:27:11+00:00" MediaCode="none" OrderTotal="0.00" PaymentType="GP" AuthCode="CP" CrossRef="" OrderDiscount="0.0000" VendorTransactionId="000000003" PaymentServiceProvider="0"/><ns1:orderitems><ns1:orderitem ProductRef="superdryjacket-Small" ProductDesc="Superdry jacket-Small" Qty="1" Rate="100.0000"/></ns1:orderitems><ns1:customer FirstName="che" Surname="such" Ad1="test street" Ad2="manchester" Country="GB" Email="che@space48.com" Telephone="0161 123 4567"><ns1:crmflags xsi:nil="true"/><ns1:giftaid xsi:nil="true"/></ns1:customer><ns1:deliveryaddress FirstName="che" Surname="such" Ad1="test street" Ad2="manchester" Country="GB"/><ns1:addinfo xsi:nil="true"/><ns1:membership xsi:nil="true"/><ns1:paymentschedule xsi:nil="true"/></ns1:order></ns1:orders></ns1:rialtoweborders><ns2:UserName>xxxxx</ns2:UserName><ns2:Password>xxxxx</ns2:Password></ns2:AddRialtoOrders></SOAP-ENV:Body></SOAP-ENV:Envelope>
 [] []`

`[2018-04-10 07:29:50] customLogHandler.INFO: <?xml version="1.0" encoding="utf-8"?><soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema"><soap:Body><AddRialtoOrdersResponse xmlns="rialto.emerymailorder.com/service.asmx"><rialtowebordersresponse ReturnStatus="1" xmlns="http://schemas.redarrow.co.uk/rialto"><orders /><errors><error OrderNo="000000003" ErrorCode="10044" ErrorMessage="Bad or missing data: customer, Pcode" /><error OrderNo="000000003" ErrorCode="10010" ErrorMessage="Order already added" /></errors></rialtowebordersresponse></AddRialtoOrdersResponse></soap:Body></soap:Envelope> [] []`

Example GetOrder output logged to file:


`[2018-04-10 07:31:38] customLogHandler.INFO: <?xml version="1.0" encoding="UTF-8"?>
<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ns1="http://schemas.redarrow.co.uk/rialto" xmlns:ns2="rialto.emerymailorder.com/service.asmx"><SOAP-ENV:Body><ns2:GetRialtoOrderDetails><ns1:rialtoorderdetails><ns1:orders><ns1:order OrderNo="000000002"/><ns1:order OrderNo="000000004"/></ns1:orders></ns1:rialtoorderdetails><ns2:UserName>xxxxx</ns2:UserName><ns2:Password>xxxxx</ns2:Password></ns2:GetRialtoOrderDetails></SOAP-ENV:Body></SOAP-ENV:Envelope>
 [] []`

`[2018-04-10 07:31:38] customLogHandler.INFO: <?xml version="1.0" encoding="utf-8"?><soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema"><soap:Body><GetRialtoOrderDetailsResponse xmlns="rialto.emerymailorder.com/service.asmx"><rialtoorderdetailsresponse ReturnStatus="2" xmlns="http://schemas.redarrow.co.uk/rialto"><orders><order OrderNo="000000002" FPOrderNo="" StatusCode="10010" DespatchDate="" CarrierRef="" CarrierName="" ConsignmentNo="" Updated=""><orderitems><orderitem OrderNo="000000002" ProductRef="product_dynamic_9" ProductDesc="Simple Product 9" DespatchDate="" CarrierName="" Updated="" /></orderitems></order><order OrderNo="000000004" FPOrderNo="" StatusCode="10010" DespatchDate="" CarrierRef="" CarrierName="" ConsignmentNo="" Updated=""><orderitems><orderitem OrderNo="000000004" ProductRef="superdryjacket-Small" ProductDesc="Superdry jacket-Small" DespatchDate="" CarrierName="" Updated="" /></orderitems></order></orders><errors /></rialtoorderdetailsresponse></GetRialtoOrderDetailsResponse></soap:Body></soap:Envelope> [] []`

File: src/Client/AddClient.php [hunk 1]
```diff
@@ -8,15 +8,20 @@
...
+use Psr\Log\LoggerInterface;
 
 class AddClient
 {
     /** @var ClientConfiguration */
     private $configuration;
 
-    public function __construct(ClientConfiguration $configuration)
+    /** @var LoggerInterface */
+    private $loggerInterface;
```
Comment (joshdifabio) [inline src/Client/AddClient.php]: The property should be called `$logger`, not `$loggerInterface`. The same goes for the constructor parameter.

This also applies to `GetClient`.

Comment (joshdifabio) [inline src/Client/AddClient.php]: The property should be called `$logger`, not `$loggerInterface`. The same goes for the constructor parameter.

This also applies to `GetClient`.

**Review by joshdifabio — CHANGES_REQUESTED** [review]
This change introduces a dependency on the PSR Log package, which must be reflected in `composer.json`. You need to run `composer require psr/log` which should add the requirement to `composer.json`.

**Pull request merged**